src/overlays/actors/ovl_En_Bb/z_en_bb.c

```diff
@@ -32,65 +105,1137 @@ const ActorInit En_Bb_InitVars = {
+                }
+                this->actor.posRot.rot.y += 0x1F40;
+            }
+            this->moveMode = BBMOVE_NORMAL;
+            this->maxSpeed = 0.0f;
+        } else {
+            this->moveMode = BBMOVE_NOCLIP;
+            this->maxSpeed = 10.0f;
+        }
+        if (this->collider.base.atFlags & 2) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_BUBLE_BITE);
+            this->collider.base.atFlags &= ~2;
+        }
+        this->actor.shape.rot.y = this->actor.posRot.rot.y;
+    } else if (Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f) == 0.0f) {
+        EnBb_FaceWaypoint(this);
+    }
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (((s32)this->skelAnime.animCurrentFrame == 0) && (Rand_ZeroOne() <= 0.1f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_BUBLE_LAUGH);
+    }
+
+    if ((this->maxSpeed != 0.0f) &&
+        (((s32)this->skelAnime.animCurrentFrame == 0) || ((s32)this->skelAnime.animCurrentFrame == 5))) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_BUBLE_MOUTH);
+    } else if (((s32)this->skelAnime.animCurrentFrame == 2) || ((s32)this->skelAnime.animCurrentFrame == 7)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_BUBLE_WING);
+    }
+}
+
+void EnBb_InitGreen(EnBb* this, GlobalContext* globalCtx) {
+    Vec3f bobOffset = { 0.0f, 0.0f, 0.0f };
+
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06000444);
+    this->moveMode = BBMOVE_NOCLIP;
+    this->actionState = 0;
+    this->bobPhase = Rand_ZeroOne();
+    this->actor.shape.rot.x = this->actor.shape.rot.z = 0;
+    this->actor.shape.rot.y = this->actor.yawTowardsLink;
+    if (this->actor.params == ENBB_GREEN_BIG) {
+        EnBb_SetWaypoint(this, globalCtx);
+        EnBb_FaceWaypoint(this);
+    }
+    Matrix_Translate(this->actor.initPosRot.pos.x, this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z,
+                     MTXMODE_NEW);
+    Matrix_RotateRPY(this->actor.posRot.rot.x, this->actor.posRot.rot.y, 0, MTXMODE_APPLY);
+    Matrix_RotateZ(this->bobPhase, MTXMODE_APPLY);
+    bobOffset.y = this->bobSize;
+    Matrix_MultVec3f(&bobOffset, &this->actor.posRot.pos);
+    this->targetActor = NULL;
+    this->action = BB_GREEN;
+    this->actor.speedXZ = 0.0f;
+    this->flameTimer = (Rand_ZeroOne() * 30.0f) + 180.0f;
+    EnBb_SetupAction(this, EnBb_Green);
+}
+
+void EnBb_SetupGreen(EnBb* this) {
+    SkelAnime_ChangeAnimDefaultRepeat(&this->skelAnime, &D_06000444);
+    this->moveMode = BBMOVE_NOCLIP;
+    this->actionState = 0;
+    this->targetActor = NULL;
+    this->action = BB_GREEN;
+    this->actor.speedXZ = 0.0f;
+    this->flameTimer = Rand_ZeroOne() * 30.0f + 180.0f;
+    this->actor.shape.rot.z = 0;
+    this->actor.shape.rot.y = this->actor.yawTowardsLink;
+    EnBb_SetupAction(this, EnBb_Green);
+}
+
+void EnBb_Green(EnBb* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
```

Comment:
naming the actions after the color is not really something i agree with, but not gonna make a big deal about documentation stuff now as we are going to focus on it more later